rainyview/src/main/java/me/samlss/view/RainyView.java

```diff
@@ -1,358 +1,435 @@
         mRemovedRainDrops = new ArrayList<>(mRainDropMaxNumber);
         mRecycler = new Stack<>();
+
     }
 
     @Override
-    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
-        super.onSizeChanged(w, h, oldw, oldh);
+    public void onRefreshed(Component component) {
+
+        int w;
+        w = component.getWidth();
+        int h;
+        h = component.getHeight();
+        int oldweight = 0;
+        int oldheight = 0;
         stop();
 
         mLeftCloudPath.reset();
         mRightCloudPath.reset();
 
-        float centerX = w / 2; //view's center x coordinate
-        float minSize = Math.min(w, h); //get the min size
-
+        //view's center x coordinate
+        float centerX;
+        centerX = w / 2;
+        //get the min size
+        float minSize = Math.min(w, h);
         //************************compute left cloud**********************
-
-        float leftCloudWidth = minSize / 2.5f; //the width of cloud
-        float leftCloudBottomHeight = leftCloudWidth / 3f; //the bottom height of cloud
-        float leftCloudBottomRoundRadius = leftCloudBottomHeight; //the bottom round radius of cloud
-
-        float rightCloudTranslateX = leftCloudWidth * 2 / 3; //the distance of the cloud on the right
-        float leftCloudEndX = (w - leftCloudWidth - leftCloudWidth * CLOUD_SCALE_RATIO / 2) / 2 + leftCloudWidth; //the left cloud end x coordinate
-        float leftCloudEndY = h / 3; //clouds' end y coordinate
-
+        //the width of cloud
+        float leftCloudWidth = minSize / 2.5f;
+        //the bottom height of cloud
+        float leftCloudBottomHeight = leftCloudWidth / 3f;
+        //the bottom round radius of cloud
+        float leftCloudBottomRoundRadius = leftCloudBottomHeight;
+        //the distance of the cloud on the right
+        float rightCloudTranslateX;
+        rightCloudTranslateX = leftCloudWidth * 2 / 3;
+        //the left cloud end x coordinate
+        float leftCloudEndX = (w - leftCloudWidth - leftCloudWidth * CLOUD_SCALE_RATIO / 2) / 2 + leftCloudWidth;
+        //clouds' end y coordinate
+        float leftCloudEndY = h / 3;
         //add the bottom round rect
-        mLeftCloudPath.addRoundRect(new RectF(leftCloudEndX - leftCloudWidth, leftCloudEndY - leftCloudBottomHeight,
-                leftCloudEndX, leftCloudEndY), leftCloudBottomHeight, leftCloudBottomRoundRadius, Path.Direction.CW);
+        mLeftCloudPath.addRoundRect(new RectFloat(leftCloudEndX - leftCloudWidth, leftCloudEndY - leftCloudBottomHeight,
+                leftCloudEndX, leftCloudEndY), leftCloudBottomHeight,
+                leftCloudBottomRoundRadius, Path.Direction.CLOCK_WISE);
 
         float leftCloudTopCenterY = leftCloudEndY - leftCloudBottomHeight;
+
         float leftCloudRightTopCenterX = leftCloudEndX - leftCloudBottomRoundRadius;
+
         float leftCloudLeftTopCenterX  = leftCloudEndX - leftCloudWidth + leftCloudBottomRoundRadius;
 
```

Comment:
Unecessary whitelines/whitespaces. Please remove. Applies to all the files...

Reply:
Done